addons/sale/models/payment.py

```diff
@@ -110,6 +110,9 @@ def _reconcile_after_transaction_done(self):
             tx._check_amount_and_confirm_order()
         # send order confirmation mail
         sales_orders._send_order_confirmation_mail()
+        # set timezone if customer not logged in
```

Comment:
Comment https://github.com/odoo/odoo/pull/100548?notification_referrer_id=NT_kwDOAPKQ7bM0NDMxMzYxOTMzOjE1ODk2ODEz#discussion_r978386684 was referring to this line too.